pages/wiki/openembedded.md

```diff
@@ -3,10 +3,245 @@ title: OpenEmbedded
+
+```
+devtool deploy-target asteroid-flashlight root@192.168.2.15
+```
+
+Now the flashlight app will show a light blue screen.  Reverting to the original version is very simple:
+
+```
+devtool undeploy-target asteroid-flashlight root@192.168.2.15
+```
+
+# Adding a New Application
+
+---
+
+Let's say you have decided to create a new application.  As described in [Creating an Asteroid app]({{rel 'wiki/creating-an-asteroid-app'}}) page, the [Asteroid-helloworld](https://github.com/AsteroidOS/asteroid-helloworld) which can be used as the basis for your own new application can be cloned and then modified.  So let's say you've done that and now you'd like to build the app, deploy it and test it.  You can, of course, follow the instructions listed on that page, but here we'll show how to do so by working within Yocto and bitbake.
+
+As mentioned above, Yocto and bitbake use *layers* and the *recipes*.  The simplest way to proceed is to use the `prepare-build.sh` file.  If you were planning, for instance on trying your app on a `sturgeon` model (a Huawei Watch), you'd navigate to the directory in which the `prepare-build.sh` file exists and run:
+
+```
+source ./prepare-build.sh sturgeon
+```
+
+This will set the necessary environment variables and place you in the `build` directory.  For these instructions, it is assumed that you have already built AsteroidOS for `sturgeon` (or your preferred test platform).  Let's say you have already cloned the `asteroid-helloworld` into a directory named `~/tools/AsteroidOS/asteroid-helloworld/`.  You can very easily create a recipe for it with `devtool`:
+
+```
+devtool add hello ~/tools/AsteroidOS/asteroid-helloworld/
+```
+
+```
+$ devtool add hello ~/tools/AsteroidOS/asteroid-helloworld/
+NOTE: Starting bitbake server...
+NOTE: Starting bitbake server...
+INFO: Please add the following line for 'LICENSE' to a 'lib/recipetool/licenses.csv' and replace `Unknown` with the license:
+84dcc94da3adb52b53ae4fa38fe49e5d,Unknown
+NOTE: Starting bitbake server...
+INFO: Recipe /home/developer/asteroid/build/workspace/recipes/hello/hello_git.bb has been automatically created; further editing may be required to make it fully functional
+```
+
+So the tool has created a recipe based on just the name we have given and the directory containing the source.  Since it seems unsure about the license and suggests that further editing might be required, let's take it's advice and edit the recipe.  The simplest way to do that is again to use `devtool`:
+
+```
+devtool edit-recipe hello
+```
+
+This will open the recipe in an editor.  It says that the license is unknown, so we change the line from `LICENSE = "Unknown"` to `LICENSE = "GPL-3.0-only"`.  Not that the `SRC_URI` is set to point to a git repository.  This is because `devtool` is smart enough to figure out that even though you gave it a local directory, it discovered that the source was actually git and made the appropriate changes.  We can now attempt to build.
```

Comment:
Both of these are likely to change in the future.  For this level of detail, I made a conscious decision to let the user discover the answer via a search engine rather than tying this site to particular sources like these.  This is, after all, only one of a hundred different kinds of such problems a user might encounter at this step and I'd rather not try to create a comprehensive "how to develop" page, but just concentrate on the items specific to `AsteroidOS` or this particular example.

Reply:
Sounds good :+1: